enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/integrations/bloom/BloomProcedures.java

```diff
@@ -57,27 +57,37 @@ public void awaitPopulation() throws IOException
 
     @Description( "Queries the bloom index for nodes" )
     @Procedure( name = "db.fulltext.bloomFulltextNodes", mode = READ )
-    public Stream<Output> bloomFulltextNodes( @Name( "terms" ) List<String> terms ) throws Exception
+    public Stream<Output> bloomFulltextNodes( @Name( "terms" ) List<String> terms, @Name( "fuzzy" ) boolean fuzzy,
+            @Name( "Require that all terms are matched" ) boolean matchAll ) throws Exception
     {
         try ( ReadOnlyFulltext indexReader = provider.getReader( BLOOM_NODES, FulltextProvider.FulltextIndexType.NODES ) )
         {
-            return queryAsStream( terms, indexReader );
+            return queryAsStream( terms, indexReader, fuzzy, matchAll );
         }
     }
 
     @Description( "Queries the bloom index for relationships" )
     @Procedure( name = "db.fulltext.bloomFulltextRelationships", mode = READ )
-    public Stream<Output> bloomFulltextRelationships( @Name( "terms" ) List<String> terms ) throws Exception
+    public Stream<Output> bloomFulltextRelationships( @Name( "terms" ) List<String> terms, @Name( "fuzzy" ) boolean fuzzy,
+            @Name( "Require that all terms are matched" ) boolean matchAll ) throws Exception
     {
         try ( ReadOnlyFulltext indexReader = provider.getReader( BLOOM_RELATIONSHIPS, FulltextProvider.FulltextIndexType.RELATIONSHIPS ) )
         {
-            return queryAsStream( terms, indexReader );
+            return queryAsStream( terms, indexReader, fuzzy, matchAll );
         }
     }
 
-    private Stream<Output> queryAsStream( List<String> terms, ReadOnlyFulltext indexReader )
+    private Stream<Output> queryAsStream( List<String> terms, ReadOnlyFulltext indexReader, boolean fuzzy, boolean matchAll )
     {
-        PrimitiveLongIterator primitiveLongIterator = indexReader.fuzzyQuery( terms.toArray( new String[0] ) );
+        PrimitiveLongIterator primitiveLongIterator;
+        if ( fuzzy )
+        {
+            primitiveLongIterator = indexReader.fuzzyQuery( matchAll, terms.toArray( new String[0] ) );
```

Comment:
This `terms.toArray( new String[0] )` is looking a bit silly. Especially when all we do with the array further down, is turning it into a `Stream`. In a future PR we could just pass the list through.